### PR TITLE
fix: Global messages disappear in the wrong order #3570

### LINKED ISSUE #3570: Global messages disappear in the wrong order
Scenario:
1 go to Consent Management,
2 check the option (appear "Consent successfully given.")
3 uncheck the option (appear "Consent successfully withdrawn.")
4 disappear "Consent successfully withdrawn."
5 disappear "Consent successfully given."

It can be easily fixed by
image
**But it is not enough**, it provides a new problem:
1 check the option (appear "Consent successfully given.")
2 uncheck the option (appear "Consent successfully withdrawn.")
3 check the option
4 disappear "Consent successfully given."
5 disappear "Consent successfully withdrawn."

---------------------------------------------------------------------

Additional found problem:
1 check the option 1 (appear "Consent successfully given.")
2 after e.g 2.8s check the option 2
3 after 0.2s "Consent successfully given." disappear

Probably to solve both issues we should create an effect for add new message, and it should remove duplicate from the existing state. Now the responsibility is in reducer, which just ignores the duplicate.

~I'm not sure, but then probably we should use an index of the newly added message (last one) like now.~

### PR DESCRIPTION
**Pull request by Pucek9**
Closes #3570

### CODE REVIEW

File: projects/core/src/global-message/store/effects/global-message.effect.spec.ts [hunk 1]
```diff
@@ -84,4 +90,25 @@ describe('GlobalMessage Effects', () => {
       );
     });
   });
+  // TODO: Test is not passing
```
Comment (Pucek9) [inline projects/core/src/global-message/store/effects/global-message.effect.spec.ts]: This test need fixes

Comment (Matejk00) [reply projects/core/src/global-message/store/effects/global-message.effect.spec.ts]: Fixed.

File: projects/core/src/global-message/store/effects/global-message.effect.spec.ts [hunk 2]
```diff
@@ -84,4 +90,26 @@ describe('GlobalMessage Effects', () => {
       );
     });
   });
+  describe('removeDuplicated$', () => {
```
Comment (RadhepS) [inline projects/core/src/global-message/store/effects/global-message.effect.spec.ts]: Can you add a test case that checks if message are removed in the correct order?

Comment (psmul) [reply projects/core/src/global-message/store/effects/global-message.effect.spec.ts]: After a long investigation, we have come to the conclusion that we're unable to write that kind of unit test. Testing effects is simple and based on defining input and outcome. As far as I know, there is no option to handle checking it in the middle. Case have been covered in unit tests above, which tests removing strictly and always first element / message from list.

**Pull request merged**